app/models/course.rb

```diff
@@ -149,6 +149,27 @@ def application_forms
       .distinct
   end
 
+  def self.find_courses_in_previous_cycle(provider)
+    previous_cycle
+        .joins(:course_options)
+        .distinct
+        .where(provider: provider)
+        .includes(%i[provider accredited_provider])
```

Comment:
Are these includes for performance reasons? I think if we refactored this into scopes we could chain the includes and keep these scope methods a bit cleaner.

If they turn out to be too complex for a scope on `Course` we should see if an existing query class is a better fit or roll our own.

Reply:
I've added a scope `with_course_options_run_by_provider` so that there is a chaining of scopes where it is called. I found that the includes didn't add or added very little to performance so I've removed